src/ArmTemplates/Common/API/Clients/Abstractions/ApiClientBase.cs

```diff
@@ -49,26 +57,52 @@ protected async Task<string> CallApiManagementAsync(string azToken, string reque
             request.Headers.UserAgent.TryParseAdd($"{Application.Name}/{Application.BuildVersion}");
 
             HttpResponseMessage response = await this.httpClient.SendAsync(request);
-            response.EnsureSuccessStatusCode();
             string responseBody = await response.Content.ReadAsStringAsync();
 
+            try
+            {
+                response.EnsureSuccessStatusCode();
```

Comment:
you have to call `EnsureSuccessStatusCode` before reading content - that is for sure.
Now you:
1) get a unstable behavior in `response.Content.ReadAsStringAsync()`
2) read it for no reason

Reply:
agree will move that into handler and later. thanks